crates/da_clients/ethereum/src/lib.rs

```diff
@@ -1,38 +1,212 @@
 #![allow(missing_docs)]
 #![allow(clippy::missing_docs_in_private_items)]
+use alloy::consensus::{
+    BlobTransactionSidecar, SignableTransaction, TxEip4844, TxEip4844Variant, TxEip4844WithSidecar, TxEnvelope,
+};
+use alloy::eips::{eip2718::Encodable2718, eip2930::AccessList, eip4844::BYTES_PER_BLOB};
+use alloy::network::{Ethereum, TxSigner};
+use alloy::primitives::{bytes, FixedBytes, TxHash, U256, U64};
+use alloy::providers::{Provider, ProviderBuilder, RootProvider};
 use alloy::rpc::client::RpcClient;
+use alloy::signers::wallet::LocalWallet;
 use alloy::transports::http::Http;
 use async_trait::async_trait;
+
 use color_eyre::Result;
+use mockall::{automock, predicate::*};
 use reqwest::Client;
-use starknet::core::types::FieldElement;
 use std::str::FromStr;
 use url::Url;
 
+use c_kzg::{Blob, KzgCommitment, KzgProof, KzgSettings};
 use config::EthereumDaConfig;
 use da_client_interface::{DaClient, DaVerificationStatus};
-
+use dotenv::dotenv;
+use std::{env, path::Path};
 pub mod config;
 pub struct EthereumDaClient {
     #[allow(dead_code)]
-    provider: RpcClient<Http<Client>>,
+    provider: RootProvider<Ethereum, Http<Client>>,
+    wallet: LocalWallet,
+    trusted_setup: KzgSettings,
 }
 
+#[automock]
 #[async_trait]
 impl DaClient for EthereumDaClient {
-    async fn publish_state_diff(&self, _state_diff: Vec<FieldElement>) -> Result<String> {
-        unimplemented!()
+    async fn publish_state_diff(&self, state_diff: Vec<Vec<u8>>) -> Result<String> {
+        dotenv().ok();
+        let provider = &self.provider;
+        let trusted_setup = &self.trusted_setup;
+        let wallet = &self.wallet;
+        let addr = wallet.address();
+
+        let (sidecar_blobs, sidecar_commitments, sidecar_proofs) = prepare_sidecar(&state_diff, trusted_setup).await?;
+        let sidecar = BlobTransactionSidecar::new(sidecar_blobs, sidecar_commitments, sidecar_proofs);
+
+        let eip1559_est = provider.estimate_eip1559_fees(None).await?;
+        let chain_id: u64 = provider.get_chain_id().await?.to_string().parse()?;
+
+        let max_fee_per_blob_gas: u128 = provider.get_blob_base_fee().await?.to_string().parse()?;
+        let max_priority_fee_per_gas: u128 = provider.get_max_priority_fee_per_gas().await?.to_string().parse()?;
+
+        let tx = TxEip4844 {
+            chain_id,
+            nonce: 0, // can be block number
+            gas_limit: 30_000_000,
+            max_fee_per_gas: eip1559_est.max_fee_per_gas.to_string().parse()?,
+            max_priority_fee_per_gas,
+            to: addr, // maybe to the L1 contract for verification??
```

Comment:
to, input, and gas_limit should probably be function arguments, if the goal is to reuse Starknet core contract, that would be its address, updateStateKZG parameters, and appropriate gas_limit to account for the spendings

Reply:
right. i think we should fill in `to` and `gas_limit` for now. `input` can be done in a refactor later on when we integrate updateState

@Mohiiit we can ideally take `to` as an input and `gas_limit` can be estimated?

Reply:
Hey @unstark, I have set the nonce by fetching it directly and used the `to` address as an input. Created an issue for the gas limit for now :)